File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        
        HashMap<String, Integer> finish = new HashMap<>();
        
        for(int i=0; i<completion.length; i++){
            finish.put(completion[i], finish.getOrDefault(completion[i],0)+1);
        }
        
        String answer = "";
        
        for(int i=0; i<participant.length; i++){
            if(finish.getOrDefault(participant[i], 0) == 0){
                answer = participant[i];
                break;
            }else{
                finish.put(participant[i], finish.get(participant[i])-1);
            }
            
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    
    var map = new Map();
    
    for(let i=0; i<completion.length; i++){
        if(map.has(completion[i])){
            map.set(completion[i], map.get(completion[i]) + 1);
        }else{
            map.set(completion[i], 1);
        }
    }
    
    var answer = '';
    
    for(let i=0; i<participant.length; i++){
        if(map.has(participant[i])){
            if(map.get(participant[i]) == 1){
                map.delete(participant[i]);
            }else{
                map.set(participant[i], map.get(participant[i]) - 1);
            }
        }else{
            answer = participant[i];
            break;
        }
    }
    
    return answer;
}